nemo/core/config/optimizers.py

```diff
@@ -79,11 +203,40 @@ class NovogradParams:
     luc_eps: float = 1e-8
 
 
-@dataclass
-class NovogradConfig:
-    """
-    Default configuration used during automagical instantiation of Novograd optimizer.
+def get_optimizer_config(name: str, **kwargs: Optional[Dict[str, Any]]) -> OptimizerParams:
     """
+    Convenience method to obtain a OptimizerParams class and partially instantiate it with optimizer kwargs.
 
-    cls: str = "novograd"
-    params: NovogradParams = NovogradParams()
+    Args:
+        name: Name of the OptimizerParams in the registry.
+        kwargs: Optional kwargs of the optimizer used during instantiation.
+
+    Returns:
+        a partially instantiated OptimizerParams
+    """
+    if name is None:
+        return kwargs
+
+    if name not in AVAILABLE_OPTIMIZER_CONFIGS:
+        raise ValueError(
+            f"Cannot resolve optimizer parameters '{name}'. Available optimizer parameters are : "
+            f"{AVAILABLE_OPTIMIZER_CONFIGS.keys()}"
+        )
+
+    scheduler_params = AVAILABLE_OPTIMIZER_CONFIGS[name]
+    scheduler_params = partial(scheduler_params, **kwargs)
+    return scheduler_params
+
+
+AVAILABLE_OPTIMIZER_CONFIGS = {
```

Comment:
Can we change the name to ```AVAILABLE_OPTIMIZER_PARAMS``` for consistency?

Reply:
or should all of these be like ```'optim_config': OptimizerConfig``` and keep ```AVAILABLE_OPTIMIZER_CONFIGS```?

Reply:
Params is cleaner than config. Note, these do not inherit from Config class (it has name in it which messes up instantiation of these classes).

Reply:
Resolved in later comment.

Reply:
Renamed these to AVAILABLE_OPTIMIZER_PARAMS to be explicit.